src/plugins/outputPlayedSong/outputPlayedSong.cpp

```diff
@@ -15,7 +18,385 @@ constexpr GH3::QbKey IntroArtistInfoText = "intro_artist_info_text";
+	0x1EBA, 0x1EBC, 0x1EBE, 0x1EC0, 0x1EC2, 0x1EC4, 0x1EC6, 0x1EC8, 0x1ECA,
+	0x1ECC, 0x1ECE, 0x1ED0, 0x1ED2, 0x1ED4, 0x1ED6, 0x1ED8, 0x1EDA, 0x1EDC,
+	0x1EDE, 0x1EE0, 0x1EE2, 0x1EE4, 0x1EE6, 0x1EE8, 0x1EEA, 0x1EEC, 0x1EEE,
+	0x1EF0, 0x1EF2, 0x1EF4, 0x1EF6, 0x1EF8, 0x1F08, 0x1F09, 0x1F0A, 0x1F0B,
+	0x1F0C, 0x1F0D, 0x1F0E, 0x1F0F, 0x1F18, 0x1F19, 0x1F1A, 0x1F1B, 0x1F1C,
+	0x1F1D, 0x1F28, 0x1F29, 0x1F2A, 0x1F2B, 0x1F2C, 0x1F2D, 0x1F2E, 0x1F2F,
+	0x1F38, 0x1F39, 0x1F3A, 0x1F3B, 0x1F3C, 0x1F3D, 0x1F3E, 0x1F3F, 0x1F48,
+	0x1F49, 0x1F4A, 0x1F4B, 0x1F4C, 0x1F4D, 0x1F59, 0x1F5B, 0x1F5D, 0x1F5F,
+	0x1F68, 0x1F69, 0x1F6A, 0x1F6B, 0x1F6C, 0x1F6D, 0x1F6E, 0x1F6F, 0x1F88,
+	0x1F89, 0x1F8A, 0x1F8B, 0x1F8C, 0x1F8D, 0x1F8E, 0x1F8F, 0x1F98, 0x1F99,
+	0x1F9A, 0x1F9B, 0x1F9C, 0x1F9D, 0x1F9E, 0x1F9F, 0x1FA8, 0x1FA9, 0x1FAA,
+	0x1FAB, 0x1FAC, 0x1FAD, 0x1FAE, 0x1FAF, 0x1FB8, 0x1FB9, 0x1FD8, 0x1FD9,
+	0x1FE8, 0x1FE9, 0x24B6, 0x24B7, 0x24B8, 0x24B9, 0x24BA, 0x24BB, 0x24BC,
+	0x24BD, 0x24BE, 0x24BF, 0x24C0, 0x24C1, 0x24C2, 0x24C3, 0x24C4, 0x24C5,
+	0x24C6, 0x24C7, 0x24C8, 0x24C9, 0x24CA, 0x24CB, 0x24CC, 0x24CD, 0x24CE,
+	0x24CF, 0xFF21, 0xFF22, 0xFF23, 0xFF24, 0xFF25, 0xFF26, 0xFF27, 0xFF28,
+	0xFF29, 0xFF2A, 0xFF2B, 0xFF2C, 0xFF2D, 0xFF2E, 0xFF2F, 0xFF30, 0xFF31,
+	0xFF32, 0xFF33, 0xFF34, 0xFF35, 0xFF36, 0xFF37, 0xFF38, 0xFF39, 0xFF3A
+};
+
+
+void ApplyWordLowercasing(std::wstring & ShoutString, int Start, int LetterCount, bool FirstLetterIsToBeCapital)
+{
+	for (int i = Start; i <= Start + LetterCount; i++)
+	{
+		if (i == Start && FirstLetterIsToBeCapital)
+		{
+			continue;
+		}
+
+		for (int j = 0; j < 9 * 74; j++)
+		{
+			if ((int)(ShoutString[i]) == Uppercase[j])
+			{
+				ShoutString[i] = Lowercase[j];
+				break;
+			}
+		}
+	}
+}
+
+void ProcessUnicodeArtistName(std::wstring & ArtistName)
+{
+	bool bDuringWord = false;
+	int LastWordStart = -1;
+
+	//fill special/whitespace chars at the end with null
+
+	for (int i = ArtistName.length() - 1; (i >= 0); i--)
+	{
+		if (ArtistName[i] != 0x0020 && ArtistName[i] != 0x0000)
+		{
+			break;
+		}
+		ArtistName[i] = 0x0000;
+	}
+
+	for (unsigned int i = 0; i < ArtistName.length(); i++)
+	{
+		if (ArtistName[i] == 0x0020 || (i == ArtistName.length() - 1))
+		{
+			if (bDuringWord) //the word ended
+			{
+				bDuringWord = false;
+				ApplyWordLowercasing(ArtistName, LastWordStart, i - LastWordStart + ((i == ArtistName.length() - 1) ? (0) : (-1)), true);
+			}
+		}
+		else
+		{
+			if (!bDuringWord)
+			{
+				bDuringWord = true;
+				LastWordStart = i;
+			}
+		}
+	}
+}
+
+void ProcessUnicodeSongName(std::wstring & SongName)
+{
+	int wordCounter = -1;
+	bool bDuringWord = false;
+	int LastWordStart = -1;
+	bool bCapitalize = true;
+
+	//fill special/whitespace chars at the end with null
+
+	for (int i = SongName.length() - 2; (i >= 0); i--)
+	{
+		if (SongName[i] != 0x0020 && SongName[i] != 0x0000)
+		{
+			break;
+		}
+		SongName[i] = 0x0000;
+	}
+
+	for (unsigned int i = 0; i < SongName.length(); i++)
```

Comment:
The very nested control flow here is difficult to follow.  Consider trying to simplify the code here